packages/intl/src/types.ts

```diff
@@ -150,7 +150,7 @@ export interface IntlFormatters<T = any, R = T> {
     values?: Record<string, PrimitiveType | FormatXMLElementFn<string, string>>,
     opts?: IntlMessageFormatOptions
   ): string
-  formatMessage(
+  formatMessage<T = any, R = T>(
```

Comment:
ideally react-intl should be able to set `T` to `ReactNode` while things like `vue-intl` or `ember-intl` can set this to something else. I don't see how we can achieve that w/ this?